src/webui/gitutils.jl

```diff
@@ -226,3 +226,44 @@ mention(u::GitLab.User) = "@$(u.username)"
 # If the registry is from the same provider, mention the user, otherwise use the URL.
 display_user(u::U) where U =
     (parentmodule(typeof(REGISTRY[].repo)) === parentmodule(U) ? mention : web_url)(u)
+
+function tagexists(forge::GitHubAPI, repo::GitHub.Repo, tag::VersionNumber)
+    result = @gf get_tags(forge, repo.owner.login, repo.name)
+
+    if result === nothing
+        @debug("Could not fetch tags")
+        return false
+    end
+
+    for t in result
+        v = split(t.ref, "/")[end]
+        if startswith(v, "v")
+            v = v[2:end]
+        end
+        v == string(tag) && return true
+    end
+    false
+end
+
+function tagexists(forge::GitLabAPI, project::GitLab.Project, tag::VersionNumber)
+    # Using Registrator's token here instead of the users. This is
```

Comment:
FYI pretty much every non-write GitLab interaction is done with the Registrator API key instead of the user's, so this comment isn't really necessary in this particular spot. I thought I had it written down in a comment somewhere but perhaps not.

Reply:
I left it as a note to myself when I have to support registration for private packages.